packages/slice-machine/src/modules/selectedSlice/reducer.ts

```diff
@@ -157,120 +155,111 @@ export const selectedSliceReducer: Reducer<
         action.payload.widgetArea,
         action.payload.newKey
       );
-
-      const newModel = { ...prevState.Model, mockConfig: updatedConfig };
-
-      return {
+      return updateTouched({
         ...prevState,
-        Model: updateTouchedAndStatus(newModel),
-      };
+        mockConfig: updatedConfig,
+      });
     }
     case getType(generateSliceScreenshotCreator): {
       if (!prevState) return prevState;
 
-      const newModel = {
-        ...prevState.Model,
+      const component: ComponentUI = {
+        ...prevState.component,
         screenshotUrls: action.payload.screenshots,
+        __status: LibStatus.Modified,
       };
 
-      return {
-        ...prevState,
-        Model: updateTouchedAndStatus(newModel),
-      };
+      return { ...prevState, component };
     }
     case getType(generateSliceCustomScreenshotCreator): {
       if (!prevState) return prevState;
       const { variationId, screenshot } = action.payload;
-      const { Model: prevModel } = prevState;
+      const prevComponent = prevState.component;
 
-      const screenshots = prevModel.variations.reduce((acc, variation) => {
-        if (variation.id === variationId) {
-          return {
-            ...acc,
-            [variationId]: screenshot,
-          };
-        }
-        if (prevModel.screenshotUrls?.[variation.id]) {
-          return {
-            ...acc,
-            [variation.id]: prevModel.screenshotUrls[variation.id],
-          };
-        }
-        return acc;
-      }, {});
-
-      const newModel = { ...prevModel, screenshotUrls: screenshots };
+      const screenshots: Record<string, ScreenshotUI> =
+        prevComponent.model.variations.reduce((acc, variation) => {
+          if (variation.id === variationId) {
+            return {
+              ...acc,
+              [variationId]: screenshot,
+            };
+          }
+          if (prevComponent.screenshotUrls?.[variation.id]) {
```

Comment:
is the `.` needed before `[selector]`?

Reply:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/69144671/179489492-bef114dd-efb9-4762-9a54-e47ce5632f8e.png">

<img width="747" alt="image" src="https://user-images.githubusercontent.com/69144671/179489603-c8189dbb-b7fb-4352-9b00-20dc9e142c5c.png">

Yes, without it typescrupt will interpret it as a ternarny `A ? B : C `